### PR TITLE
Update postgres-baseline

### PR DESCRIPTION
**Pull request by FLiPp3r90**
This PR includes:
* Add some security control checks
* Disable older Postgresql versions
* Check pg_hba.conf with pg_hba_conf resource
* Replace attributes with inputs
* Some little modifications

### CODE REVIEW

File: controls/postgres_spec.rb [hunk 1]
```diff
@@ -129,10 +63,8 @@
   impact 1.0
   title 'Run one postgresql instance per operating system'
   desc 'Only one postgresql database instance must be running on an operating system instance (both physical HW or virtualized).'
-  pg_command = 'postgres'
-  pg_command = 'postmaster' if os.redhat? && os.release.include?('6.')
-  describe processes(pg_command) do
-    its('entries.length') { should eq 1 }
+  describe command('ps aux | awk /\'bin\/postmaster\'/ | wc -l') do
```
Comment (rndmh3ro) [inline controls/postgres_spec.rb:66]: Does this mean that the process is called `postmaster` on all OS'es now? Have you had the possibility to test it?

File: controls/postgres_spec.rb [hunk 2]
```diff
@@ -304,3 +247,62 @@
     its('log_line_prefix') { should eq '%t %u %d %h' }
   end
 end
+
+control 'postgres-17' do
+  impact 1.0
+  title 'Grants should not be assigned to public'
+  desc 'Grants should not be assigned to public to avoid issues with tenant separations.'
+  describe postgres_session(USER, PASSWORD).query("SELECT COUNT(*) FROM   information_schema.table_privileges WHERE  grantee = 'PUBLIC' AND table_schema NOT LIKE 'pg_catalog' AND table_schema NOT LIKE 'information_schema';") do
```
Comment (rndmh3ro) [inline controls/postgres_spec.rb:255]:
```suggestion
  describe postgres_session(USER, PASSWORD).query("SELECT COUNT(*) FROM information_schema.table_privileges WHERE grantee = 'PUBLIC' AND table_schema NOT LIKE 'pg_catalog' AND table_schema NOT LIKE 'information_schema';") do
```

File: controls/postgres_spec.rb [hunk 3]
```diff
@@ -304,3 +247,62 @@
...
+    its('output') { should eq '0' }
+  end
+end
+
+control 'postgres-18' do
+  impact 1.0
+  title 'Grants should not be assigned with grant option privilege'
+  desc 'Grants should not be assigned with grant option exept postgresql admin superuser.'
+  describe postgres_session(USER, PASSWORD).query("SELECT COUNT(is_grantable) FROM   information_schema.table_privileges WHERE grantee NOT LIKE 'postgres' AND is_grantable = 'YES';") do
```
Comment (rndmh3ro) [inline controls/postgres_spec.rb:264]:
```suggestion
  describe postgres_session(USER, PASSWORD).query("SELECT COUNT(is_grantable) FROM information_schema.table_privileges WHERE grantee NOT LIKE 'postgres' AND is_grantable = 'YES';") do
```

**Review by rndmh3ro — COMMENTED** [review]
Looks good! I only have some minor remarks.

File: controls/postgres_spec.rb (hunk 1 above)
Comment (FLiPp3r90) [reply controls/postgres_spec.rb:66]: Good point, not realy sure. I will look at this.

**Pull request closed without merging**